### PR TITLE
Fix CALLs

### PR DESCRIPTION
**Pull request by axic**
Closes #246 (replaces).

### CODE REVIEW

**Comment by axic** [comment]
Commits regarding the "2300 call stipend" can be removed if https://github.com/ewasm/hera/pull/219 is merged.

File: wasm/generateInterface.js [hunk 1]
```diff
@@ -281,7 +338,7 @@ function generateManifest (interfaceManifest, opts) {
     (i64.store (i32.add (get_global $sp) (i32.const ${spOffset * 32 + 8 * 2})) (i64.const 0))`
 
       if (!op.async) {
-        call += '(drop (call $bswap_m128 (i32.add (i32.const 32)(get_global $sp))))'
+        call += '(drop (i32.add (i32.const 32)(get_global $sp)))'
```
Comment (axic) [inline wasm/generateInterface.js]: This line is useless, it is a no-op.

In the previous case `bswap_m128` operates on memory, so it wasn't a no-op.

File: wasm/generateInterface.js [hunk 2]
```diff
@@ -292,7 +349,7 @@ function generateManifest (interfaceManifest, opts) {
       }
 
       call += `)
-    (drop (call $bswap_m160 (i32.add (get_global $sp) (i32.const ${spOffset * 32}))))
+    (drop (i32.add (get_global $sp) (i32.const ${spOffset * 32})))
```
Comment (axic) [inline wasm/generateInterface.js]: Same applies here.

File: wasm/generateInterface.js [hunk 3]
```diff
@@ -223,13 +223,33 @@ function generateManifest (interfaceManifest, opts) {
         // the wasm memory offset is a new item on the EVM stack
         spOffset++
         call += `(i32.add (get_global $sp) (i32.const ${spOffset * 32}))`
+      } else if (input === 'i64' && opcode === 'CALL') {
+        // i64 param for CALL is the gas
+        // add 2300 gas subsidy
+        // for now this only works if the gas is a 64-bit value
+        // TODO: use 256-bit arithmetic
+        /*
```
Comment (axic) [inline wasm/generateInterface.js]: Why is this kept as a comment?

Comment (cdetrio) [reply wasm/generateInterface.js]: it will be relevant when we simplify the EEI and move the ugly gas logic here into evm2wasm

**Pull request merged**